unstructured/ingest/cli/cmds/pinecone.py

```diff
@@ -0,0 +1,66 @@
+import typing as t
+from dataclasses import dataclass
+
+import click
+
+from unstructured.ingest.cli.interfaces import (
+    CliMixin,
+)
+from unstructured.ingest.interfaces import BaseConfig
+
+
+@dataclass
+class PineconeCliWriteConfig(BaseConfig, CliMixin):
+    api_key: str
+    index_name: str
+    environment: str
+    batch_size: int
+    num_processes: int
+
+    @staticmethod
+    def get_cli_options() -> t.List[click.Option]:
+        options = [
+            click.Option(
+                ["--api-key"],
+                required=True,
+                type=str,
+                help="API key used for authenticating to a Pinecone instance.",
+                envvar="PINECONE_API_KEY",
+                show_envvar=True,
+            ),
+            click.Option(
+                ["--index-name"],
+                required=True,
+                type=str,
+                help="The name of the pinecone index to connect to.",
+            ),
+            click.Option(
+                ["--environment"],
+                required=True,
+                type=str,
+                help="The environment where the index lives. Eg. 'gcp-starter' or 'us-east1-gcp'",
+            ),
+            click.Option(
+                ["--batch-size"],
+                default=50,
+                type=int,
+                help="Number of records per batch",
+            ),
+            click.Option(
```

Comment:
This should be pulled from the overall pipeline config, rather than a new one here.

Reply:
I think user might want to set different number of processes for partitioning versus uploading to a destination, if for example destination has rate limits. User might want to have magnitudes of higher number of threads for partitioning compared to uploading content. What do you think?

Reply:
If we want to support both, it should default to what is set by the processor config and overwrite with this value if one is provided.

Reply:
I still don't understand why the writer's number of processes should default to processor's number of processes. 

If a user didn't specify a `num_processes` for the writer, why do we opt them into the same number of processes as in for the processor? 

User will have to opt out of that number whenever they want a different value if we do it that way. Whereas if we have a separate argument, it'll behave in an intuitive way where that separate writer argument has its own default value.

Is this thread a technical concern such as stated in https://github.com/Unstructured-IO/unstructured/pull/1774/#discussion_r1399725686 (another thread down below), rather than a user related concern? To quote that thread:

> This should leverage the same num_processes being set in the ProcessorConfig. Actually not sure if this is causing a duplicate key error in the CLI itself...

If our concern is this quote, the test already checks if cli works with this `num_processes` argument for the writer, I cannot see any issue with that. To test even further I now made them have different values with [different number of processes for processor and writer in test](https://github.com/Unstructured-IO/unstructured/pull/1774/commits/940f72d5fe66129a7b5ed50c380a5a6560557ce5) .

Please let me know if I am missing anything here.